tools/test_history.py

```diff
@@ -44,7 +44,12 @@ class Case(TypedDict):
     skipped: bool
 
 
-class Suite(TypedDict):
+# The original format of the s3 reports did not have this filename field. This maintains backwards-compatibility
+class SuiteMeta(TypedDict, total=False):
+    filename: str
```

Comment:
`filename` would not be available for all the historical data, is it?
```suggestion
    filename: Optional[str]
```

Reply:
I think Optional[str] annotation meant `filename` can be either a string or None. but still it requires to be explicitly set (e.g. `suiteMeta.filename = None` is still needed otherwise error will thrown). 
based on https://mypy.readthedocs.io/en/latest/more_types.html#totality, seems like the `total=False` part makes the key filename optional